pygfx/renderers/wgpu/_pipeline.py

```diff
@@ -740,7 +741,7 @@ def get_shader_module(self, device, source):
         # todo: cache more objects, like pipelines once we figure out how to clean things up
 
         assert isinstance(source, str)
-        key = source  # or hash(code)
+        key = hashlib.sha1(source.encode()).hexdigest()
```

Comment:
Well, this is ridiculous and makes no sense. Is it possible that `source` is actually a subclass of string or something else manipulated by Jinja?